packages/@aws-cdk/aws-appsync/lib/private.ts

```diff
@@ -12,6 +15,76 @@ export enum SchemaMode {
   CODE = 'CODE',
 };
 
+/**
+ * Generates an addition to the schema
+ *
+ * ```
+ * prefix name interfaces directives {
+ *   field
+ *   field
+ *   ...
+ * }
+ * ```
+ *
+ * @param prefix the prefix for this additon (type, interface, enum, input, schema)
+ * @param name the name for this addition (some additions dont need this [i.e. schema])
+ * @param interfaceTypes the interface types if this is creating an object type
+ * @param directives the directives for this type
+ * @param fields the fields to reduce onto the addition
+ */
+export interface SchemaAdditionOptions {
+  readonly prefix: string;
+  readonly name?: string;
+  readonly interfaceTypes?: InterfaceType[];
+  readonly directives?: Directive[];
+  readonly fields: string[];
+}
+
+/**
+ * Utility function to generate interfaces for object types
+ *
+ * @param interfaceTypes the interfaces this object type implements
+ */
+function generateInterfaces(interfaceTypes?: InterfaceType[]): string {
+  if (!interfaceTypes || interfaceTypes.length === 0) return '';
+  return interfaceTypes.reduce((acc, interfaceType) =>
+    `${acc} ${interfaceType.name},`, ' implements').slice(0, -1);
+}
+
+/**
+ * Utility function to generate directives
+ *
+ * @param directives the directives of a given type
+ * @param delimiter the separator betweeen directives
+ * @default - ' '
```

Comment:
not needed here because we don't expose it. but we also normally try to provide a description of the default behaviour - i.e. cover what `' '` means as part of the doc string